protocols/v2/noise-sv2/src/signature_message.rs

```diff
@@ -24,27 +24,35 @@ impl From<[u8; 74]> for SignatureNoiseMessage {
 }
 
 impl SignatureNoiseMessage {
-    pub fn verify(self, pk: &XOnlyPublicKey) -> bool {
-        let now = SystemTime::now()
-            .duration_since(SystemTime::UNIX_EPOCH)
-            .unwrap()
-            .as_secs() as u32;
-        if self.valid_from <= now && self.not_valid_after >= now {
-            let secp = Secp256k1::verification_only();
-            let (m, s) = self.split();
-            let m = Message::from_hashed_data::<sha256::Hash>(&m[0..10]);
-            let s = match Signature::from_slice(&s) {
-                Ok(s) => s,
-                _ => return false,
-            };
-            secp.verify_schnorr(&s, &m, pk).is_ok()
-        } else {
-            false
+    pub fn verify(self, pk: &XOnlyPublicKey, authority_pk: &Option<XOnlyPublicKey>) -> bool {
+        match authority_pk {
+            Some(authority_pk) => {
```

Comment:
not need to change it. just to let you know, in this case idiomatic rust would be to use 
```rust
if let Some(auth_pk) = auth_pk {
   ....
}
```

Reply:
https://github.com/stratum-mining/stratum/pull/752/commits/b4a4387f647831f57ac8f75ce8ebdd8c8d46983a